### DISCUSSION TITLE
List of special keycodes

### DISCUSSION THREAD

**Discussion by brunodavi** [florisboard/florisboard#1496] [Q&A]
Good afternoon, I would like to know the list of special keycodes, I searched the nas wiki, I didn't find it, in the `.json` files I found many but they still weren't the ones I was looking for and finally I found `KeyCodes.kt` but the codes were wrong.

<details>
<summary>Examples</summary>

![KeyCode.kt](https://user-images.githubusercontent.com/48166556/148701470-394e05c0-b3a2-4ef1-a664-6e80a410fe09.jpg)

> This is the codes I tried to use to create an extension.

![termux_extesion_example](https://user-images.githubusercontent.com/48166556/148701547-76ba15ce-364b-4e25-8fa8-ea52970c4c2a.jpg)

> It would be similar to what I have in termux

![example_extention_json](https://user-images.githubusercontent.com/48166556/148701573-bd0cb806-0131-46f1-8750-d8d0820675dd.jpg)

> This is the extension I would use to test the codes
> And try to create my extension

![20220109_173752](https://user-images.githubusercontent.com/48166556/148701693-9b52dae2-e5ad-473b-8130-59da85e5f79c.jpg)

> But unfortunately it didn't work.

</details>

> NOTE: I had also tested if it works with the default keycode, but it doesn't work either.

So I come humbly to ask, the keycode list

Anyone who can help me would be very grateful

**Comment by patrickgold** [comment]
The list of key codes you found in the master branch are indeed the correct key codes. However there's a catch: you are still using 0.3.13 and these special key codes (everything below zero) were changed during 0.3.14 beta development. In the current release 0.3.14-beta08 the above key codes will apply fine (although not all codes have an icon assigned atm).

**Reply by brunodavi** [reply]
I got it, thanks
